### PR TITLE
Examples/Tests file clean up.

### LINKED ISSUE #845: Add eslint overrides to allow us to clean up eslint comments in doc and test files
# Feature Request

## Description
In terra-core, we recently worked on a PR to add eslint overrides to clean up the eslint comments needed in doc and test files. 
https://github.com/cerner/terra-core/pull/2589

This issue is for completing the same work in this repo.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fixes #845

### CODE REVIEW

File: packages/terra-abstract-modal/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ Changelog
 Unreleased
 ----------
 ### Changed
-Updated `keycode-js` from `v1.0.4` to `v2.0.1`
+* Cleaned up imports in examples and test files
+* Updated `keycode-js` from `v1.0.4` to `v2.0.1`
```
Comment (ShettyAkarsh) [inline packages/terra-abstract-modal/CHANGELOG.md]: Is this part of your change ?

Comment (saket2403) [reply packages/terra-abstract-modal/CHANGELOG.md]: It was a merge conflict in some changelogs

File: packages/terra-slide-panel/src/terra-dev-site/doc/slide-panel/SlidePanel.1.doc.jsx [hunk 2]
```diff
@@ -28,7 +27,7 @@ import SlidePanelLarge from '../../test/slide-panel/SlidePanelLarge.test';
 import SlidePanelFullscreen from '../../test/slide-panel/SlidePanelFullscreen.test';
 import SlidePanelFill from '../../test/slide-panel/SlidePanelFill.test';
 import SlidePanelNoFill from '../../test/slide-panel/SlidePanelNoFill.test';
-/* eslint-enabled import/no-webpack-loader-syntax, import/first, import/extensions, import/no-unresolved, import/no-duplicates */
+
```
Comment (avinashg1994) [inline packages/terra-slide-panel/src/terra-dev-site/doc/slide-panel/SlidePanel.1.doc.jsx]: nit: I am just nit picking here, but you could remove this extra space just for consistency, like as in the other files.

File: package.json [hunk 3]
```diff
@@ -26,16 +26,30 @@
...
-      "import/no-extraneous-dependencies": [
-        "error",
-        {
-          "devDependencies": [
-            "./packages/*/src/terra-dev-site/**/*.jsx"
-          ]
+    "overrides": [
+      {
+        "files": [
+          "**/*/terra-dev-site/**/*.jsx"
```
Comment (supreethmr) [inline package.json]: are we not disabling this rules for components and  jest tests .. ?

Comment (saket2403) [reply package.json]: No only examples and test files.

**Pull request merged**